alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteMonitorAction.kt

```diff
@@ -69,22 +66,21 @@ class TransportDeleteMonitorAction @Inject constructor(
     val xContentRegistry: NamedXContentRegistry
 ) : HandledTransportAction<DeleteMonitorRequest, DeleteResponse>(
     DeleteMonitorAction.NAME, transportService, actionFilters, ::DeleteMonitorRequest
-) {
+),
+    SecureTransportAction {
 
-    @Volatile private var filterByEnabled = AlertingSettings.FILTER_BY_BACKEND_ROLES.get(settings)
+    @Volatile override var filterByEnabled = AlertingSettings.FILTER_BY_BACKEND_ROLES.get(settings)
 
     init {
-        clusterService.clusterSettings.addSettingsUpdateConsumer(AlertingSettings.FILTER_BY_BACKEND_ROLES) { filterByEnabled = it }
+        listenFilterBySettingChange(clusterService)
```

Comment:
Since this is a method defined in an interface and is being used as an `init`, should we make the method `final` to prevent unexpected behavior or is the expectation that this method be overridden in the future for more complicated inheritance hierarchies of `SecureTransportAction`?

Reply:
I think this should be a var since this is a setting that can change overtime

Reply:
I cant make `listenFilterBySettingChange` final since it is in an interface.

Reply:
I don't think the `AlertingSettings.FILTER_BY_BACKEND_ROLES` will be changing too often.

If the user needs to specify that listener in the `init` manually for any child class anyway, it might be better to have them explicitly state it rather than leave `listenFilterBySettingChange` open to override since it could be misleading.

Alternatively, we could make the class abstract to make the method `final`. Just something to consider for longevity, I don't think it's a blocker to merge.